game_event/views.py

```diff
@@ -108,45 +134,36 @@ def game_event(request, id):
         'in_event': in_event,
         'event_players': event_players
         }
-    return render(request, 'game_event/game-event.html', context)
-
-
-def join_event(request, id):
-    try:
-        GameEvent.objects.get(pk=id)
-    except GameEvent.DoesNotExist:
-        return render(request, 'game_event/join-event.html', {'id': id, 'event_exists': False})
-
-    return render(request, 'game_event/join-event.html', {'id': id, 'event_exists': True})
-
-
-def remove_from_event(request, id):
-    try:
-        event = GameEvent.objects.get(pk=id)
-    except GameEvent.DoesNotExist:
-        return render(request, 'game_event/result.html', {'event_exists': False})
-    player = Player.objects.get(user=request.user)
-    GameEventPlayer.objects.get(game_event=event, player=player).delete()
-    return render(request, 'game_event/result.html', {'event_exists': True})
+    return context
 
 
 def process_answer_game_event(request, id):
     try:
         event = GameEvent.objects.get(pk=id)
     except GameEvent.DoesNotExist:
-        return render(request, 'game_event/result.html', {'event_exists': False})
-
+        return redirect(f"/game-events/{id}", {})
+    content = {}
     player = Player.objects.get(user=request.user)
     ball_responsible = request.POST.get('answer') == 'yes'
-    content = {}
-    num_of_players_in_event = len(GameEventPlayer.objects.filter(game_event=event))
-    if num_of_players_in_event < event.max_number_of_players:
-        GameEventPlayer.objects.create(game_event=event, player=player, ball_responsible=ball_responsible).save()
-        in_event = GameEventPlayer.objects.filter(game_event=event, player=player).exists()
-        content["in_event"] = in_event
-        content["reached_max"] = False
+    if GameEventPlayer.objects.filter(game_event=event, player=player):
+        messages.error(request, "Already in event!")
+        return redirect(f"/game-events/{id}", add_content(event, player))
+
+    is_event_full = event.is_event_full()
+    is_event_time_available = player.is_event_time_available(event.time)
+
+    error_messages = []
+    if is_event_full:
```

Comment:
I think this validation should be on the even page not after we click on the join